crates/fluvio-socket/src/error.rs

```diff
@@ -12,9 +12,7 @@ pub enum SocketError {
 
 impl From<IoError> for SocketError {
     fn from(err: IoError) -> Self {
-        SocketError::Io {
-            source: err,
-            msg: "".to_string(),
-        }
+        let msg = err.to_string();
+        SocketError::Io { source: err, msg }
```

Comment:
this could be shortend without intermediate var
```
SocketError::Io { source: err, msg: err_to_string() }
```

Reply:
Hmm, that gives me a borrow error because err is moved to source.

Reply:
Oh right.  silly rust borrow checker limitation